contract/models/contract.py

```diff
@@ -229,44 +289,62 @@ def _get_recurring_create_invoice_domain(self, contract=False):
         return domain
 
     @api.multi
-    def _get_lines_to_invoice(self, date_ref=False):
+    def _get_lines_to_invoice(self, use_contract_as_ref=False, date_ref=False):
+        """
+        This method fetches and returns the lines to invoice on the contract
+        (self).
+        :param use_contract_as_ref: if True, use the contract's date of next
+        invoice as reference date
+        :param date_ref: if set, use it as reference date
+        (if use_contract_as_ref is not True). Otherwise, take the date of today
+        :return: contract lines (account.analytic.invoice.line recordset)
+        """
         self.ensure_one()
         if not date_ref:
             date_ref = fields.Date.context_today(self)
+        if use_contract_as_ref:
+            date_ref = self.recurring_next_date
```

Comment:
@sbidoul @sbejaoui I had to restore the functionality that allows to base the search of lines to invoice on the contract's next invoice date because it broke the `recurring_create_invoice` button and unit tests.

Reply:
@ThomasBinsfeld can you point me to the test that broke?

Reply:
@sbidoul https://github.com/OCA/contract/pull/207/files#diff-8b8162cf7b037d9ef8f62464d0c4c7c5R1397 because it uses the button that was broken.

Reply:
@ThomasBinsfeld I think it's not the best fix to do. IMO, the caller of _get_lines_to_invoice should pass recurring_next_date as date_ref. What do you think?

Reply:
@ThomasBinsfeld that was the point of my previous comment there: https://github.com/sbejaoui/contract/pull/4#discussion_r242900895

Reply:
@sbidoul Like that? https://github.com/sbejaoui/contract/commit/28a3acdfca0f69cb15a54556b98179f751611821

The default date_ref must be today but i'm still not sure of what should be the invoice date (currently `contract.recurring_next_date`).